### ISSUE TITLE
Is there a way to make haskell-updater unneccesary?

### ISSUE THREAD

**Issue by crocket** [gentoo-haskell/haskell-updater#8]
haskell-updater seems like a workaround. Ideally, emerge shouldn't need haskell-updater to upgrade a system with haskell packages.

**Comment by trofi** [comment]
`haskell-updater` solves the problem that gentoo's package manager does not provide (yet):
- https://bugs.gentoo.org/449094
- https://github.com/gentoo-haskell/gentoo-haskell/issues/199

Tl;DR is: haskell ABI is too fine-grained to express in terms of Gentoo's dependency specification in ebuilds.

**Comment by crocket** [comment]
Does EAPI=6 solve this issue?

**Comment by trofi** [comment]
No. There is no syntax or semantics agreed to provide required functionality.
